src/app/file-manager/models/item.ts

```diff
@@ -19,13 +18,22 @@ export class Item {
   transferred: number;
   remaining: number;
   bucketName: string;
+  workspaceName: string;
   prefix: string;
   delimiter: string;
   open: boolean;
+  selected: boolean;
+  indeterminate: boolean;
```

Comment:
Using a enum to represent the three different selection states instead of having two boolean fields could simplify the code, as it would make it easer to reset and read this value in the code. The two booleans can represent four states but only three are used, as far as I can tell. Since the values of each relies on the other, an enum should make the code simpler.

Reply:
selected and indeterminate are two different properties. Each value is handled by a different checkbox property as well. Not sure to understand this comment. Changing this implies a lot of changes in the code and will require to go through QA again.  All design was done with two different properties in mind.

Reply:
OK, maybe I don't understand what these fields mean. I thought 'selected' is true when the item is selected, and 'indeterminate' is true when the item is partially selected (e.g. it's a folder with only some of its contents selected). They are both false when the item is neither selected or partially selected. Therefore there is no use case where they are both true, and we only have three states. Looking at the code, it looked like they are almost always both updated at the same time, another sign that they are representing a single (composite) value. If this is all true, than creating an enum with three states would be a valid replacement.